instrumentation/rails/lib/opentelemetry/instrumentation/rails/railtie.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Rails
+      # The Instrumentation class contains logic to detect and install the Rails
+      # instrumentation, while this Railtie is used to conventionally instrument
+      # the Rails application through its initialization hooks
+      class Railtie < ::Rails::Railtie
+        initializer 'opentelemetry.before_initialize' do |app|
+          app.middleware.insert_after(
+            ActionDispatch::RequestId,
```

Comment:
I don't really have any strong opinions on **where** we should put the tracing middleware by default.

Reply:
Our Shopify tracing middleware stuck it here because we wanted to record the request ID as an attribute. We're not doing that here (but maybe we should?).